sapi/fpm/tests/gh12385.phpt

```diff
@@ -0,0 +1,46 @@
+--TEST--
+GH-12385 (flush with fastcgi does not force headers to be sent)
```

Comment:
> I confirmed in my testing that two FCGI packets will be sent: one for the header and one for the content, which is what is requested.

The feature request is about sending headers when `flush()` is called. I am not sure exactly what you meant by the cited sentence, but if the response buffer/body is empty, no (empty) packet for data should be sent.

> If we have only few bytes to flush then nginx waits for more bytes to send in the TCP packet.

Am I understanding you right, headers are always sent with `flush()` (even with empty body) but small body is not sent even if `flush()` is called?

related discussion https://serverfault.com/questions/488767/how-do-i-enable-php-s-flush-with-nginxphp-fpm

Reply:
Based on https://stackoverflow.com/questions/20018803/flush-output-buffer-in-apache-nginx-setup#23171223 and https://stackoverflow.com/questions/4870697/php-flush-that-works-even-in-nginx it seems a header from php can be sent to inform the nginx to not cache the response. FYI only.